### DISCUSSION TITLE
How to invalidate cache in webpack 5?

### DISCUSSION THREAD

**Discussion by BUPTlhuanyu** [webpack/webpack#15098] [Q&A] [answered]
I have some loaders to analyze less file to collect data from `theme.js`, and i use filesystem cache and webpack-dev-middleware.

when the file `theme.js` has changed but the less files have not changed, the loaders will not analyze the less file anymore because of the cache, maybe the memory cache of webpack-dev-middleware

whatever, how to watch the `theme.js` and invalidate the cache, so that the loaders will handle these less files to collect data?

**Comment by alexander-akait** [comment, marked as answer]
Using https://webpack.js.org/configuration/cache/#cachebuilddependencies

**Reply by BUPTlhuanyu** [reply]
> Using https://webpack.js.org/configuration/cache/#cachebuilddependencies

I used it before, It's seems not work

![image](https://user-images.githubusercontent.com/27213108/147949957-5d510116-77a7-40ae-9e9a-ce6e3d61d767.png)

**Reply by BUPTlhuanyu** [reply]
i use the api of webpack-dev-middleware to rebuild and return the result like this
![image](https://user-images.githubusercontent.com/27213108/147950461-44c7eb9a-0e44-404a-8ceb-fd283c85d4c7.png)

**Reply by alexander-akait** [reply]
If you collection information from files into strings (without file modifications) you should use them in https://webpack.js.org/configuration/cache/#cacheversion
